src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,192 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { Actions, DispatchContext, Keys } from '../Store';
+import { deleteData, updateC, updateT } from '../../api/todos';
+
+interface Props {
+  todo: Todo,
+  loaderActive: boolean,
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  loaderActive,
+}) => {
+  const dispatch = useContext(DispatchContext);
+  const { id, title, completed } = todo;
+  const [currentTitle, setCurrentTitle] = useState(title);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isUpdating, setIsUpdating] = useState(loaderActive);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleInputToogle = () => {
+    setIsUpdating(true);
+    updateC({ id, completed: !completed })
```

Comment:
not a good naming for a function
```suggestion
    updateTodo({ id, completed: !completed })
```